test/lint/lint-includes.sh

```diff
@@ -57,6 +57,7 @@ EXPECTED_BOOST_INCLUDES=(
     boost/filesystem.hpp
     boost/filesystem/fstream.hpp
     boost/multi_index/hashed_index.hpp
+    boost/multi_index/member.hpp
```

Comment:
I'm importing this file so I can use the default key extractors in the `indexed_rebroadcast_set` multi_index: 1. https://github.com/bitcoin/bitcoin/pull/21061/files#diff-eccf6b88ae8b612dbdbdb92c110b66e4e7e3ffa665ccdc1acf08093aadc66b82R49 2. https://github.com/bitcoin/bitcoin/pull/21061/files#diff-eccf6b88ae8b612dbdbdb92c110b66e4e7e3ffa665ccdc1acf08093aadc66b82R55

The contents of the header file can be found [here](https://www.boost.org/doc/libs/1_71_0/boost/multi_index/member.hpp).

If we'd rather not import another boost header, I can avoid by writing custom key extractors.

Reply:
I think it's fine to add a boost multi_index header dependency, it's the only submodule of boost that we really need and for which there is no replacement on the horizon. So just use what you need IMO, re-implementing anything from multi-index without a good plan would be a waste of time. IMO.